python/cudf/cudf/core/column/decimal.py

```diff
@@ -366,6 +366,20 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
     else:
         raise NotImplementedError()
 
+    # potentially reduce scale and precision
+    if precision > cudf.Decimal128Dtype.MAX_PRECISION:
```

Comment:
Can we make it so that we don't go through the top level `cudf` namespace here? It can make it harder to resolve circular import issues later on down the line.